doc/modules/ensemble.rst

```diff
@@ -940,375 +843,371 @@ based on permutation of the features.
+on randomized :ref:`decision trees <tree>`: the RandomForest algorithm
+and the Extra-Trees method. Both algorithms are perturb-and-combine
+techniques [B1998]_ specifically designed for trees. This means a diverse
+set of classifiers is created by introducing randomness in the classifier
+construction.  The prediction of the ensemble is given as the averaged
+prediction of the individual classifiers.
 
-These histogram-based estimators can be **orders of magnitude faster**
-than :class:`GradientBoostingClassifier` and
-:class:`GradientBoostingRegressor` when the number of samples is larger
-than tens of thousands of samples.
+As other classifiers, forest classifiers have to be fitted with two
+arrays: a sparse or dense array X of shape ``(n_samples, n_features)``
+holding the training samples, and an array Y of shape ``(n_samples,)``
+holding the target values (class labels) for the training samples::
 
-They also have built-in support for missing values, which avoids the need
-for an imputer.
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> X = [[0, 0], [1, 1]]
+    >>> Y = [0, 1]
+    >>> clf = RandomForestClassifier(n_estimators=10)
+    >>> clf = clf.fit(X, Y)
 
-These fast estimators first bin the input samples ``X`` into
-integer-valued bins (typically 256 bins) which tremendously reduces the
-number of splitting points to consider, and allows the algorithm to
-leverage integer-based data structures (histograms) instead of relying on
-sorted continuous values when building the trees. The API of these
-estimators is slightly different, and some of the features from
-:class:`GradientBoostingClassifier` and :class:`GradientBoostingRegressor`
-are not yet supported, for instance some loss functions.
+Like :ref:`decision trees <tree>`, forests of trees also extend to
+:ref:`multi-output problems <tree_multioutput>`  (if Y is an array
+of shape ``(n_samples, n_outputs)``).
 
-.. topic:: Examples:
+Random Forests
+--------------
 
- * :ref:`sphx_glr_auto_examples_inspection_plot_partial_dependence.py`
+In random forests (see :class:`RandomForestClassifier` and
+:class:`RandomForestRegressor` classes), each tree in the ensemble is built
+from a sample drawn with replacement (i.e., a bootstrap sample) from the
+training set.
 
-Usage
------
+Furthermore, when splitting each node during the construction of a tree, the
+best split is found either from all input features or a random subset of size
+``max_features``. (See the :ref:`parameter tuning guidelines
+<random_forest_parameters>` for more details).
 
-Most of the parameters are unchanged from
-:class:`GradientBoostingClassifier` and :class:`GradientBoostingRegressor`.
-One exception is the ``max_iter`` parameter that replaces ``n_estimators``, and
-controls the number of iterations of the boosting process::
+The purpose of these two sources of randomness is to decrease the variance of
+the forest estimator. Indeed, individual decision trees typically exhibit high
```

Comment:
```suggestion
the forest estimator. Individual decision trees typically exhibit high
```
I don't think we need the indeed, and I think "indeed" is something to use when you are agreeing with something that has already been said. Maybe "In fact" is better but I would just remove the indeed and not add anything.